source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -179,6 +179,22 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
     // We can only insert trailers at the end of data, so keep track of this value
     // until then.
     grpc_status_ = grpcStatusFromHeaders(headers);
+
+    // This is a header-only response, and we should prepend the gRPC frame
+    // header directly.
+    if (end_stream) {
```

Comment:
I think we should make the `end_stream && withhold_grpc_frames_`. When `withhold_grpc_frames_` is false, the upstream is responsible for grpc framing and I think we should forward it's invalid encoding rather than paper over the error.

Reply:
Thanks and agreed! I didn't realize that that was the intent of `withhold_grpc_frames_`.